src/synth/fluid_voice.c

```diff
@@ -1168,6 +1197,74 @@ int fluid_voice_modulate_all(fluid_voice_t* voice)
   return FLUID_OK;
 }
 
+/** legato update functions --------------------------------------------------*/
+/* Updates voice portamento parameter
+ *
+ * @voice voice the synthesis voice
+ * @fromkey the begining pitch of portamento.
+ * @tokey the ending pitch of portamento.
+ *
+ * The function calculates pitch offset and increment, then the parameters
+ * are send to the dsp 
+*/
+void fluid_voice_update_portamento (fluid_voice_t* voice, int fromkey,int tokey)
+									
+{
+	fluid_channel_t* channel= voice->channel;
+	fluid_real_t PitchBeg = fluid_voice_calculate_pitch(voice,fromkey);
+	fluid_real_t PitchEnd = fluid_voice_calculate_pitch(voice,tokey);
+	fluid_real_t pitchoffset = PitchBeg - PitchEnd;
+	/* Increment number is function of PortamentoTime (ms)*/
+	unsigned int countinc = (int)(((fluid_real_t)voice->output_rate * 
+					0.001f *
+			        (fluid_real_t)fluid_channel_portamentotime(channel))  /
+					(fluid_real_t)FLUID_BUFSIZE  +0.5);
```

Comment:
Why are you casting this to `int` when assigning it to an `unsigned int`?

Reply:
This a writing error and this should be cast  `(unsigned int)`.